### PR TITLE
Using a BaseModelInterface instead of a BaseModel

### PR DESCRIPTION
**Pull request by namelivia**
As propsed by @carlosafonso in the issue #11 the `BaseModel` is no longer present and it is replaced by a `BaseModelInterface` that declares the `getValidationRules()` function.
The test for the `BaseModel` are also gone.

Now if a project uses a Model has to implement the interface like this:

``` php
<?php

namespace App\Http\Models;

use Illuminate\Database\Eloquent\Model;
use Afonso\LvCrud\Models\BaseModelInterface;

class Foo extends Model implements BaseModelInterface
{
    public $fillable = ['name'];
    public function rules(Request $request)
    {
        return [
            'name' => 'required'
        ];
    }
    public function getValidationRules() {
        return [
            'name' => 'required'
        ];
    }
}
```

### CODE REVIEW

File: src/Models/BaseModelInterface.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+namespace Afonso\LvCrud\Models;
+
+interface BaseModelInterface
+{
+	public function getValidationRules();
```
Comment (carlosafonso) [inline src/Models/BaseModelInterface.php]: This function needs documenting, as well as the interface.

I think the old name `BaseModel` wasn't quite right. How about naming this interface `CrudModelInterface`? Developers might already have some other sort of base model.

Comment (namelivia) [inline src/Models/BaseModelInterface.php]: Agree

**Comment by carlosafonso** [comment]
`CrudController` now expects the related model to implement this new interface. We should enforce this expectation somehow to avoid errors. Or at least we should raise our very own error.

**Comment by carlosafonso** [comment]
Rebase your commits before merging please.

**Pull request merged**